### PR TITLE
Add option to set Authorization: header

### PR DESCRIPTION
**Pull request by vbraun**
I'm using JWT to authenticate to the server which is typically communicated by an "Authorization:" header. This patch adds an option so I can just do

``` html
        <file-upload authorization="Bearer my_token_goes_here"></file-upload>
```

### CODE REVIEW

**Comment by winhowes** [comment]
I think this may be more helpful to generalize it to any sort of request header. I'll go ahead and merge this if you wouldn't mind changing null to empty string, and make the generalization sometime this week

File: file-upload.html [hunk 1]
```diff
@@ -215,6 +215,14 @@
...
         type: Boolean,
         value: true
+      },
+
+      /**
+       * Authorization header
+      */
+      authorization: {
+        type: String,
+        value: null,
```
Comment (winhowes) [inline file-upload.html:225]: Can this be an empty string instead of null

**Comment by vbraun** [comment]
I thought about whether there is a use case for any other header but couldn't come up with one; Do you have any? Imho thats the key question. If there is then one should just have one way to specify headers, presumably as Object key/values.

About the null, if you are into legalese then it is explicitly allowed to have a header whose value is the empty byte sequence. You might want to be able to distinguish that from no header at all. Though I didn't really do that either and just test whether it is truthy at the end.

**Comment by winhowes** [comment]
I've seen some custom headers.  And yeah, the only reason I was thinking of changing null to empty string was just because it's type is string (though I agree it doesn't really matter when it's being evaluated as truthy)

**Pull request merged**